### PR TITLE
apple script proof of concept

### PR DESCRIPTION
**Pull request by christian-schulze**
- [ ] Updated related documentations
- [ ] Added the change to the Changelog

Hi there,
I'd like to share with you this proof-of-concept Apple Script integration, where you can call an Event in your javascript config via an Apple Script.

__NOTE__ This PR is not in any shape to be merged, and I would like your guidance as to how best to implement this feature if you consider it worthwhile pursuing.

The following is an example javascript config:
```js
const event = new Event('dispatch', function (event, data) {
  const text = data
    ? 'Event from AppleScript with data: ' + data
    : 'Event from AppleScript';

  Modal.build({ duration: 5, text }).show();
});
```

And the following is an example Apple Script:
```applescript
tell application "Phoenix"
  dispatch
  dispatch event with "testing testing 1 2 3"
end tell
```

You could also run the following from the command line:
```bash
osascript -e 'tell application "Phoenix" to dispatch'
```

Properly implemented, this could address the following issues:
- https://github.com/kasper/phoenix/issues/215
- https://github.com/kasper/phoenix/issues/140

So what do you think, is this worth pursuing?

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,6 +7,7 @@
     "npm": ">=7.14.0"
   },
   "devDependencies": {
+    "at-least-node": "1.0.0",
```
Comment (christian-schulze) [inline package.json:10]: __NOTE__ i needed to install this otherwise the javascript part of the build process would fail

**Comment by kasper** [comment]
@christian-schulze Hey, cool idea! I’ll check this with thought once I have more time.

**Comment by kasper** [comment]
@christian-schulze This looks really cool and is actually a pretty simple solution to the problem! 👍 I wonder though, should we investigate instead using the Shortcuts app and command line tools since Apple is planning to deprecate AppleScript and Automator?

**Comment by christian-schulze** [comment]
All good, I've since started hacking on all things linux, but still use a mac in my day job, so happy to come back to this. Can you provide a bit of detail about the Shortcuts app and other command line tools? I've not heard of this yet...

**Comment by kasper** [comment]
@christian-schulze Sure! Here’s some references: https://support.apple.com/en-gb/guide/shortcuts-mac/apdf22b0444c/mac and https://support.apple.com/en-gb/guide/shortcuts-mac/apd455c82f02/5.0/mac/12.0.

**Pull request closed without merging**